rfcs/react-components/convergence/semantic-shared-colors.md

```diff
@@ -0,0 +1,379 @@
+
+> Do not use CSS variables directly
+>
+> ⚠ Never use theme CSS variables directly! The CSS variables implementation of the theme is internal to the library. We might eventually decide to change the variable names, hash them or even use direct values instead of some variables. Always use the tokens to access the theme.
+
+Add "status" slots to `cranberry`, add `orange` as a status color - 15 new tokens added.
+
+Add semantic color tokens which are mapped to the current shared colors. For example, `colorPaletteDangerBackground1` is mapped to `--colorPaletteRedBackground1`. `Danger` is still mapped to the old `red` color, but the token name is now semantic. Still not changed to `cranberry`.
+
+Update the sources in the library to use the new colors.
+
+Update the mapping from old colors to the new colors.
+
+```ts
+const tokens = {
+  // new tokens (note: naming is not final) - the mapping is not 1:1
+  colorStatusDangerBackground1: 'var(--colorPaletteCranberryBackground1)',
+  colorStatusWarningBackground1: 'var(--colorPaletteOrangeBackground1)',
+
+  colorPaletteCranberryBackground1: 'var(--colorPaletteCranberryBackground1)',
+  colorPaletteOrangeBackground1: 'var(--colorPaletteOrangeBackground1)',
+};
+
+const webLightTheme = {
+  colorPaletteCranberryBackground1: 'HEX_VALUE',
+  colorPaletteOrangeBackground1: 'HEX_VALUE',
+};
+```
+
+#### 4B: Add CSS variables for new status tokens, map old tokens to the status CSS variables
+
+The variant 4 proposes to keep `--colorPaletteRedBackground1`, create new token `colorPaletteDangerBackground1` and map it to the red CSS variable.
+We can also implement this the other way around - create `colorPaletteDangerBackground1` mapped 1:1 to `--colorPaletteDangerBackground1` and re-map the legacy tokens `colorPaletteRedBackground1` to the status CSS variables.
+
+```ts
+const tokens = {
+  // new tokens (note: naming is not final)
+  colorStatusDangerBackground1: 'var(--colorStatusDangerBackground1)',
+  colorStatusWarningBackground1: 'var(--colorStatusWarningBackground1)',
+
+  // current tokens - the mapping CHANGED, it is not 1:1
+  colorPaletteCranberryBackground1: 'var(--colorStatusDangerBackground1)',
+  colorPaletteOrangeBackground1: 'var(--colorStatusWarningBackground1)',
+};
+
+const webLightTheme = {
+  colorStatusDangerBackground1: 'HEX_VALUE',
+  colorStatusWarningBackground1: 'HEX_VALUE',
+};
+```
+
+#### Pros and Cons
+
+- 👍 Semantic tokens without any unnecessary tokens.
+- 👍 Controlled partner upgrade from old hex values to new values, but coupled with library upgrade.
+- 👎 This is currently not supported by token pipeline - would require either changes there or custom implementation on library side.
+- 👎 Semantic and status tokens are still coupled - the only way for the application to override `danger` is to override `red` (which will affect other parts of the UI).
```

Comment:
This might confuse partners, as they will want to override the `danger`, it will just not work.